pkg/opensearch/response_parser.go

```diff
@@ -86,29 +87,187 @@ func (rp *responseParser) getTimeSeries(configuredFields es.ConfiguredFields) (*
+						transformedResourceAttributes = append(transformedResourceAttributes, map[string]interface{}{"key": k, "value": v})
+					}
+					if transformedResourceAttributes != nil {
+						doc["serviceTags"] = transformedResourceAttributes
+					}
+				}
+
+				continue
+			}
+			// determine if we need to add error flags to display icons in trace panel
+			spanHasError := withKeysToObj["events"] != nil && utils.SpanHasError(withKeysToObj["events"].([]interface{}))
+			if k == "span" {
+				spanAttributes, ok := v.(map[string]interface{})["attributes"].(map[string]interface{})
+				if spanAttributes != nil && ok {
+					transformedSpanAttributes := []map[string]interface{}{}
+					for k, v := range spanAttributes {
+						transformedSpanAttributes = append(transformedSpanAttributes, map[string]interface{}{"key": k, "value": v})
+					}
+					if spanHasError {
+						transformedSpanAttributes = append(transformedSpanAttributes, map[string]interface{}{"key": "error", "value": true})
+
+					}
+					if transformedSpanAttributes != nil {
+						doc["tags"] = transformedSpanAttributes
+					}
+				}
+
+				continue
+			}
+			if k == "events" {
+				spanEvents := []map[string]interface{}{}
+				stackTraces := []string{}
+				if len(v.([]interface{})) > 0 {
+					for _, event := range v.([]interface{}) {
+						eventObj := event.(map[string]interface{})
+						timeStamp, err := utils.TimeFieldToMilliseconds(eventObj["time"])
+						if err != nil {
+							return backend.ErrDataResponse(500, err.Error())
+						}
+						spanEvents = append(spanEvents, map[string]interface{}{"timestamp": timeStamp, "fields": []map[string]interface{}{{"key": "name", "value": eventObj["name"]}}})
+
+						// get stack traces if error event
+						attributes, ok := eventObj["attributes"].(map[string]interface{})
+						if ok {
+							errorValue := attributes["error"]
+							if errorValue != nil {
+								stackTraces = append(stackTraces, fmt.Sprintf("%s: %s", eventObj["name"], attributes["error"]))
+							} else {
+								continue
+							}
+						}
+					}
+					if spanHasError {
+						doc["stackTraces"] = stackTraces
+					}
+
+					doc["logs"] = spanEvents
+					continue
+				}
+			}
+			doc[k] = v
+		}
+
+		if hit["fields"] != nil {
+			source, ok := hit["fields"].(map[string]interface{})
+			if ok {
+				for k, v := range source {
+					doc[k] = v
+				}
+			}
+		}
+		for key := range doc {
+			propNames[key] = true
+		}
+		docs[hitIdx] = doc
+	}
+
+	sortedPropNames := sortPropNames(propNames, []string{})
+
+	fields := processDocsToDataFrameFields(docs, sortedPropNames, false)
+
+	frame := data.NewFrame("", fields...)
+	if frame.Meta == nil {
+		frame.Meta = &data.FrameMeta{}
+	}
+	frame.Meta.PreferredVisualization = data.VisTypeTrace
+
+	queryRes.Frames = data.Frames{frame}
+	return queryRes
+}
+func processLogsResponse(res *es.SearchResponse, limitString string, configuredFields es.ConfiguredFields, queryRes backend.DataResponse) backend.DataResponse {
```

Comment:
I don't think `limitString` is being used anywhere.

Reply:
Fixed it after rebase!